libs/ImpSpec/src/Test/ImpSpec/Internal.hs

```diff
@@ -97,6 +98,12 @@ deriving instance (Eq (ImpSpecEnv t), Eq (ImpSpecState t)) => Eq (ImpInit t)
 deriving instance (Ord (ImpSpecEnv t), Ord (ImpSpecState t)) => Ord (ImpInit t)
 deriving instance (Show (ImpSpecEnv t), Show (ImpSpecState t)) => Show (ImpInit t)
 
+impInitEnvL :: Lens' (ImpInit t) (ImpSpecEnv t)
+impInitEnvL = lens impInitEnv $ \x y -> x {impInitEnv = y}
+
+impInitStateL :: Lens' (ImpInit t) (ImpSpecState t)
+impInitStateL = lens impInitState $ \x y -> x {impInitState = y}
```

Comment:
We not adding any lenses to ImpSpec because that would result in an unnecessary dependency in a reusable package.

Moreover, we don't need to add a lens everytime we have a record. If a lens is going to be used only once (like in this PR) it is ok to just use regular record syntax.